lib/experimental/Navigation/Header/Breadcrumbs/index.tsx

```diff
@@ -59,14 +68,82 @@ interface BreadcrumbsProps {
 }
 
 export default function Breadcrumbs({ breadcrumbs }: BreadcrumbsProps) {
+  const [visibleCount, setVisibleCount] = useState(2)
+  const containerRef = useRef<HTMLDivElement>(null)
+
+  useEffect(() => {
+    const container = containerRef.current
+    if (!container) return
+
+    const updateVisibleBreadcrumb = () => {
+      if (!containerRef.current || breadcrumbs.length <= 2) {
+        setVisibleCount(breadcrumbs.length)
+        return
+      }
+
+      const containerWidth = containerRef.current.offsetWidth
+      const itemWidth = 150
+      const dropdownWidth = 50
```

Comment:
Do we use these "magic" numbers in another place? Maybe we could have them declared outside of the `Breadcrumbs` component as a global const variable